crates/client/mempool/src/genesis/mod.rs

```diff
@@ -0,0 +1,471 @@
+
+                    DevnetPredeployedContract { secret: key, pubkey: pubkey.scalar(), balance, address }
+                })
+                .collect(),
+        )
+    }
+
+    pub fn build(mut self, chain_config: &ChainConfig) -> anyhow::Result<UnverifiedFullBlock> {
+        self.initial_balances.to_storage_diffs(chain_config, &mut self.initial_storage);
+
+        Ok(UnverifiedFullBlock {
+            header: UnverifiedHeader {
+                parent_block_hash: None,
+                sequencer_address: chain_config.sequencer_address.to_felt(),
+                block_timestamp: SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .expect("Current time is before unix epoch!")
+                    .as_secs(),
+                protocol_version: chain_config.latest_protocol_version,
+                l1_gas_price: GasPrices {
+                    eth_l1_gas_price: 5,
+                    strk_l1_gas_price: 5,
+                    eth_l1_data_gas_price: 5,
+                    strk_l1_data_gas_price: 5,
+                },
+                l1_da_mode: dp_block::header::L1DataAvailabilityMode::Blob,
+            },
+            state_diff: StateDiff {
+                storage_diffs: self.initial_storage.as_state_diff(),
+                deprecated_declared_classes: self.declared_classes.as_legacy_state_diff(),
+                declared_classes: self.declared_classes.as_state_diff(),
+                deployed_contracts: self.deployed_contracts.as_state_diff(),
+                replaced_classes: vec![],
+                nonces: vec![],
+            },
+            declared_classes: self.declared_classes.into_loaded_classes()?,
+            ..Default::default()
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::l1::MockL1DataProvider;
+    use crate::{block_production::BlockProductionTask, Mempool, MempoolProvider};
+    use crate::{transaction_hash, L1DataProvider};
+    use blockifier::abi::abi_utils::get_fee_token_var_address;
+    use blockifier::abi::sierra_types::next_storage_key;
+    use dc_block_import::{BlockImporter, Validation};
+    use dc_db::DeoxysBackend;
+    use dp_block::header::L1DataAvailabilityMode;
+    use dp_block::{BlockId, BlockTag};
+    use dp_convert::felt_to_u128;
+    use dp_receipt::{Event, ExecutionResult, FeePayment, InvokeTransactionReceipt, PriceUnit, TransactionReceipt};
+    use dp_transactions::broadcasted_to_blockifier;
+    use rstest::{fixture, rstest};
+    use starknet_core::types::{
+        BroadcastedDeclareTransaction, BroadcastedDeployAccountTransaction, BroadcastedInvokeTransaction,
+        BroadcastedInvokeTransactionV3, BroadcastedTransaction, DeclareTransactionResult,
+        DeployAccountTransactionResult, InvokeTransactionResult, ResourceBounds, ResourceBoundsMapping,
+    };
+
+    use std::sync::Arc;
+
+    struct PreparedDevnet {
```

Comment:
I think that's pretty cool that this live in its own crate under a feature flag ?

Reply:
this for tests only, you mean you want that for appchain developers?